### PR TITLE
[QASM import] u3, u2, id gates

### LINKED ISSUE #1775: Qasm output should include identity gates

### PR DESCRIPTION
**Pull request by balopat**
Adds support for `u3`, `u2`, `id` gates in QASM import and improved testing by parametrized tests for arg and parameter error paths and expanded consistency check using the currently supported gates.  

This is PR number 8 for the piecemeal reimplementation of #1602.

3 more PRs are planned: 
- single qubit gates - x,y,z,h,t,tdg,s,sdg
- two qubit gates - cx (=`CX`),cy,cz,ch, swap
- three qubit gates - ccx, cswap

Fixes #1775.

### CODE REVIEW

File: cirq/contrib/qasm_import/qasm_test.py [hunk 1]
```diff
@@ -11,12 +11,29 @@
...
 
 # TODO as we grow the language, we'll add more complex examples here
 def test_consistency_with_qasm_output():
-    circuit1 = cirq.Circuit()
+    a, b, c, d = [cirq.NamedQubit('q_{}'.format(i)) for i in range(4)]
+    circuit1 = cirq.Circuit.from_ops(
+        cirq.Rx(np.pi / 2).on(a),
+        cirq.Ry(np.pi / 2).on(b),
+        cirq.Rz(np.pi / 2).on(b),
+        cirq.IdentityGate(1).on(c),
```
Comment (balopat) [inline cirq/contrib/qasm_import/qasm_test.py]: TODO: look into qasm output for Identity

Comment (balopat) [reply cirq/contrib/qasm_import/qasm_test.py]: Actually - `u3(0,0,0)` was the current output for a 1 qubit IdentityGate which is not bad, but for two qubits:

```
// Generated from Cirq v0.6.0.dev

OPENQASM 2.0;
include "qelib1.inc";


// Qubits: [q1, q2]
qreg q[2];


// Gate: I(2)
u3(pi*1.0,0,pi*0.5) q[0];
u3(0,0,0) q[1];
rx(pi*0.5) q[0];
cx q[0],q[1];
rx(pi*0.5) q[0];
ry(pi*0.5) q[1];
cx q[1],q[0];
rx(pi*-0.5) q[1];
rz(pi*0.5) q[1];
cx q[0],q[1];
u3(pi*1.0,0,pi*0.5) q[0];
u3(0,0,0) q[1];

```

Comment (balopat) [reply cirq/contrib/qasm_import/qasm_test.py]: with the new commits, the two qubit version is: 

```
// Generated from Cirq v0.6.0.dev

OPENQASM 2.0;
include "qelib1.inc";


// Qubits: [0, 1]
qreg q[2];


id q[0];
id q[1];
```

File: cirq/circuits/qasm_output.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
 from cirq import ops, linalg, protocols, value
 
 
+@value.value_equality(unhashable=True, approximate=True)
```
Comment (Strilanc) [inline cirq/circuits/qasm_output.py]: Why unhashable?

Comment (balopat) [reply cirq/circuits/qasm_output.py]: yeah, it is not strictly necessary, I can fix it in the next one. 
 
I am still unsure what is the use of a __hash__ function in the approximate equality case? In my understanding, checking equality on these classes is not useful due to the floating point nature of the underlying values. And having the default `_value_equality_hash` implementation is more expensive than a `None`...

Comment (Strilanc) [reply cirq/circuits/qasm_output.py]: approximate=True does not change the behavior of `__eq__`, it adds a method `_approx_eq_`.

**Pull request merged**